compiler/resolution/generics.cpp

```diff
@@ -83,28 +83,28 @@ explainInstantiation(FnSymbol* fn) {
         if (first)
           first = false;
         else
-          len += sprintf(msg+len, ", ");
+          len += snprintf(msg+len, sizeof(msg)-len, ", ");
```

Comment:
This is what I was suggesting we do in `get_version` too (line, msg+len, buff_size-len).

Reply:
Though funnily enough this still isn't memory safe (due to underflows methinks)

Reply:
What case would make this unsafe?

Reply:
Ah, wait, nevermind. The return of `snprintf` would never exceed `sizeof(msg)-len`, so it should never go negative. It's safe.